openfecli/commands/gather.py

```diff
@@ -85,40 +86,85 @@ def legacy_get_type(res_fn):
         return 'complex'
 
 
-def _get_ddgs(legs):
+def _generate_bad_legs_error_message(set_vals, ligpair):
+    expected_rbfe = {'complex', 'solvent'}
+    expected_rhfe = {'solvent', 'vacuum'}
+    maybe_rhfe = bool(set_vals & expected_rhfe)
+    maybe_rbfe = bool(set_vals & expected_rbfe)
+    if maybe_rhfe and not maybe_rbfe:
+        msg = (
+                "This appears to be an RHFE calculation, but we're "
+                f"missing {expected_rhfe - set_vals} runs for the "
+                f"edge with ligands {ligpair}."
+            )
+    elif maybe_rbfe and not maybe_rhfe:
+        msg = (
+            "This appears to be an RBFE calculation, but we're "
+            f"missing {expected_rbfe - set_vals} runs for the "
+            f"edge with ligands {ligpair}."
+        )
+    elif maybe_rbfe and maybe_rhfe:
+        msg = (
+            "Unable to determine whether this is an RBFE "
+            f"or an RHFE calculation. Found legs {set_vals} "
+            f"for ligands {ligpair}. Those ligands are missing one "
+            f"of: {(expected_rhfe | expected_rbfe) - set_vals}."
+        )
+    else:  # -no-cov-
+        # this should never happen
+        msg = (
+            "Something went very wrong while determining the type "
+            f"of RFE calculation. For the ligand pair {ligpair}, "
+            f"we found legs labelled {set_vals}. We expected either "
+            f"{expected_rhfe} or {expected_rbfe}."
+        )
+
+    msg += (
+        "\n\nYou can force partial gathering of results, without "
+        "problematic edges, by using the --allow-partial flag of the gather "
+        "command. Note that this may cause problems with predicting "
+        "absolute free energies from the relative free energies."
+    )
+    return msg
+
+
+def _get_ddgs(legs, error_on_missing=True):
     import numpy as np
     DDGs = []
     for ligpair, vals in sorted(legs.items()):
+        set_vals = set(vals)
         DDGbind = None
         DDGhyd = None
         bind_unc = None
         hyd_unc = None
 
-        if 'complex' in vals and 'solvent' in vals:
+        if set_vals == {'complex', 'solvent'}:
```

Comment:
this is working a little differently to before.  If I ran all three of vacuum, solvent and complex then this wouldn't now trigger either condition

looking at the original code, because it's an `elif` it also won't trigger *both* branches either

can we change this to intersections with the above cases rather than equality?

Reply:
Take a look at [8c643fe](https://github.com/OpenFreeEnergy/openfe/pull/588/commits/8c643fe6a52b2d636446bf136a2a79a10521e3b6). However, the output table will be nonsense if you do that.